src/test/java/decodex/parser/ParserTest.java

```diff
@@ -39,11 +44,23 @@ void getUserArguments_userInputWithOneSpaceSeparator_expectArrayOfOneArgument()
         String testInput = "test data";
         String[] expectedOutput = new String[]{"data"};
 
-        String[] arguments = parser.getUserArguments(testInput);
+        String[] arguments = parser.getUserArgumentsAsArray(testInput);
```

Comment:
If you are removing the `getUserArgumentsAsArray`, perhaps replace this test for one using `getUserArgumentsAsString`?